frontend/app/layout.tsx

```diff
@@ -18,9 +20,23 @@ export default function RootLayout({
 }>) {
   return (
     <html lang="en">
+      <Script id="theme">
+        {`  if (
+              localStorage.theme === "dark" ||
+              (!("theme" in localStorage) &&
+                window.matchMedia("(prefers-color-scheme: dark)").matches)
+            ) {
+              localStorage.theme = "dark";
+              document.documentElement.classList.add("dark");
+            } else {
+              localStorage.theme = "light";
+              document.documentElement.classList.remove("dark");
+            }`}
+      </Script>
```

Comment:
Should this code be duplicated here? I think the exact same code exists in `ThemeSwitch.tsx` as the `switchThemes` function.

Reply:
Its for initialization purposes